api/proto/teleport/scim/v1/scim_service.proto

```diff
@@ -80,13 +92,43 @@ message UpdateSCIMResourceRequest {
   // Resource is a representation of the updated resource that the server needs
   // to conform with
   Resource resource = 2;
+
+  // Meta holds information about the original SCIM/HTTP request that triggered
+  // this internal SCIM operation request.
+  RequestMetadata meta = 3;
 }
 
 // DeleteSCIMResourceRequest describes a request to delete a SCIM-mananged
 // resource
 message DeleteSCIMResourceRequest {
   // Target is the owner, type and ID if the resource targeted by the request.
   RequestTarget target = 1;
+
+  // Meta holds information about the original SCIM/HTTP request that triggered
+  // this internal SCIM operation request.
+  RequestMetadata meta = 2;
+}
+
+// RequestMetadata holds information about the original SCIM/HTTP request
+message RequestMetadata {
+  // ID is a request ID to be included in all logging related to the handling of
+  // this request
+  string id = 1;
+
+  // SourceAddress is the IP address of the client that made this request
+  string source_address = 2;
+
+  // UserAgent is the UserAgent of the enclosing SCIM request
+  string user_agent = 3;
+
+  // Method is the HTTP method of the enclosing SCIM request
+  string method = 4;
+
+  // Path is the HTTP request path for the enclosing SCIM request
+  string path = 5;
+
+  // Body is a representation of the original request JSON body
+  google.protobuf.Struct body = 6;
```

Comment:
What benefits do we gain by emitting SCIM audit events in auth rather than in the proxy?

If there aren’t clear advantages, I’d like to propose emitting the audit events directly from the SCIM proxy handler. At that point we already have everything we need from the HTTP request/response (path, method, payload, status), and we avoid duplicating information like body is parsed payload in the `CreateSCIMResourceRequest` etc.

Currently only SCIM proxy role is able to call auth SCIM handlers so from my understending emitting audit event from proxy is much simpler. 

WDYT ?

Reply:
I did look into that, and definitely have no objection to adding it there; I just didn't see any examples of emitting events from the proxy (at least in `e`) and went with the pattern.

I will put this on hold and try emitting the proxy again.